server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/routes/TasksRoutes.java

```diff
@@ -181,4 +189,46 @@ private TaskId getTaskId(Request req) {
                 .haltError();
         }
     }
+
+    private Duration getTimeout(Request req) {
+        try {
+            Optional<String> requestTimeout = Optional.ofNullable(req.queryParams("timeout"))
+                .filter(parameter -> !parameter.isEmpty());
+
+            requestTimeout.ifPresent(timeout ->
+                Preconditions.checkState(!timeout.replaceAll(" ", "").startsWith("-"), "Timeout should be positive"));
+
+            Duration timeout = requestTimeout
+                .map(rawString -> DurationParser.parse(rawString, ChronoUnit.SECONDS))
+                .orElse(MAXIMUM_AWAIT_TIMEOUT);
+
+            Preconditions.checkState(timeout.compareTo(MAXIMUM_AWAIT_TIMEOUT) <= 0, "Timeout should not exceed one year");
```

Comment:
A precondition at the end of a method look weird. Moving it to a ensureXxx method would solve this.

Reply:
I did not find an example, can you give me one please?

Reply:
See See https://github.com/linagora/james-project/pull/2653/commits/2f10137516980ab1b8c39b7a20722cc31d0ca2e8

Reply:
Thank you